ladybug/location.py

```diff
@@ -13,15 +15,18 @@ class Location(object):
         longitude: Location longitude between -180 (west) and 180 (east) (Default: 0).
         time_zone: Time zone between -12 hours (west) and 12 hours (east) (Default: 0).
         elevation: A number for elevation of the location.
-        station_id: Id of the location if the location is represnting a weather station.
+        station_id: ID of the location if the location is represnting a weather station.
         source: Source of data (e.g. TMY, TMY3).
+        building_id: ID of the building if the data comes from a simulation.
```

Comment:
Ah I didn't realize that you were thinking that DataCollections would be like that.  In that case, shouldn't we be removing the location object from the DataCollection, then?  Since it seems that you have realized the location EPW location object has a different purpose than holding the city, building, and zone information that get displayed along with DataCollection visualizations?  By this same logic, I guess all of the energy simulation results that would be converted to DataCollections don't need to store the latitude and longitude on each and every DataCollection Header.  So I guess that these are two different objects?  One that stores the location information of an EPW file and another that stores the names of cities, weather stations, buildings and zones to help identify where a DataCollection comes from?

Reply:
@mostaphaRoudsari ,

Are you saying that this metadata should just be an empty dictionary by default, or should it have pre-defined keys for 'city', 'building', 'zone'?  I worry a bit that, if it doesn't have some structure to it, it may become more difficult to build things on top of it in the future (like all of the future capabilities that I imagine matching E+ results to HBZones using zone names).  I can see situations arising where we come up with different keys in this dictionary to represent the same thing if there isn't an agreed-upon nomenclature for the keys.

If it should have pre-defined attributes, shouldn't it then be its own object? (I was imagining an object in the same file as header.py)

I was also imagining that this object would be very different from the location object in that this metadata object would only store the city name, building name, and zones name (and possibly a point # if the data comes from an analysisgrid).  The Location object would have none of these properties and just hold latitude, longitude, time zone, and elevation.

Reply:
I removed the location object from the DataCollection and replaced it with an extensible metadata dictionary.